ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Protocol/TPraos.hs

```diff
@@ -30,11 +39,18 @@ import           Ouroboros.Consensus.Shelley.Protocol.Abstract (ProtoCrypto,
                      ProtocolHeaderSupportsLedger (..),
                      ProtocolHeaderSupportsProtocol (..), ShelleyHash (..),
                      ShelleyProtocol, ShelleyProtocolHeader, protocolHeaderView)
+import           Ouroboros.Consensus.Util (whenJust)
 
 type instance ProtoCrypto (TPraos c) = c
 
 type instance ShelleyProtocolHeader (TPraos c) = SL.BHeader c
 
+data TPraosEnvelopeError
+  = ChainPredicateFailure ChainPredicateFailure
+  | InvalidCheckpoint -- TODO args
```

Comment:
same comment as other new constructor name

Reply:
Renamed to CheckpointMismatch in 8c48082